app/rag_system.py

```diff
@@ -321,8 +336,10 @@ def rebuild(self):
         Rebuild the embeddings for the knowledge base. This should be called whenever the knowledge base is updated.
         """
         print("Rebuilding embeddings for the knowledge base...")
-        self.knowledge_base = self.load_knowledge_base()  # Reload the knowledge base
-        self.doc_embeddings = self.rebuild_embeddings()  # Rebuild the embeddings
+        knowledge_base = self.load_knowledge_base()  # Reload the knowledge base
+        self.doc_embeddings = self.rebuild_embeddings(
+            knowledge_base
+        )  # Rebuild the embeddings
```

Comment:
It looks like `rebuild_embeddings` already updates `self.doc_embeddings`. 
```suggestion
        self.rebuild_embeddings(
            knowledge_base
        )  # Rebuild the embeddings
```